alcs-api/src/providers/typeorm/migrations/1660859890136-active_days_calculations.ts

```diff
@@ -0,0 +1,123 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class activeDaysCalculations1660859890136 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      CREATE OR REPLACE FUNCTION get_weekday_count (start_date DATE, end_date DATE)
+          RETURNS int
+          LANGUAGE plpgsql
+          AS $$
+      DECLARE
+          weekday_count integer;
+      BEGIN
+          SELECT
+              SUM(
+                  CASE WHEN extract(dow FROM s)
+                  IN(0, 6) THEN
+                      0
+                  ELSE
+                      1
+                  END) INTO weekday_count
+          FROM
+              generate_series(start_date, end_date, '1 day'::interval) s;
+          RETURN weekday_count;
+      END;
+      $$;
+    `);
+
+    await queryRunner.query(`
+      CREATE OR REPLACE FUNCTION get_holiday_count (start_date DATE, end_date DATE)
+          RETURNS int
+          LANGUAGE plpgsql
+          AS $$
+      DECLARE
+          holiday_count integer;
+      BEGIN
+          SELECT
+              COUNT(UUID) INTO holiday_count
+          FROM
+              holiday_entity
+          WHERE
+              holiday_entity. "day" BETWEEN start_date
+              AND end_date;
+          RETURN holiday_count;
+      END;
+      $$;
+    `);
+
+    await queryRunner.query(`DROP function calculate_paused_time;`);
```

Comment:
why do you need to drop function here?

Reply:
Because the return type changed, create or replace will error out and you must DROP it first